src/application-page-status/ApplicationPageStatus.jsx

```diff
@@ -0,0 +1,71 @@
+import { useRef, useContext, useLayoutEffect } from 'react';
+import PropTypes from 'prop-types';
+import Button from 'terra-button';
+import uuidv4 from 'uuid/v4';
+import ApplicationPageStatusContext from './ApplicationPageStatusContext';
+
+const propTypes = {
+  /**
+   * An array of objects containing terra-button properties. A key attribute is required for each object.
+   * This array is used to render buttons in the bottom section.
+   * Example:`[{ text: 'Button 1', key: 1, size: 'medium', variant: 'action', onClick: onClickFunction}]`
+   */
+  // eslint-disable-next-line react/forbid-foreign-prop-types
+  buttonAttrs: PropTypes.arrayOf(PropTypes.shape(Button.propTypes)),
+
+  /**
+   * The descriptive text, displayed under the title.
+   */
+  message: PropTypes.string,
+
+  /**
+   * The title displayed under the glyph. Variants contain default titles that can be overridden by this prop.
+   */
+  title: PropTypes.string,
+
+  /**
+   * Sets the glyph and title using a pre-baked variant. One of the following: `no-data`,
+   * `no-matching-results`, `not-authorized`, or `error`
+   */
+  variant: PropTypes.oneOf(['no-data', 'no-matching-results', 'not-authorized', 'error']),
+};
+
+const defaultProps = {
+  buttonAttrs: [],
+  message: undefined,
+  title: undefined,
+  variant: undefined,
+};
+
+const ApplicationPageStatus = (props) => {
+  const {
+    buttonAttrs,
+    message,
+    title,
+    variant,
+  } = props;
+  const id = useRef(uuidv4());
+  const applicationPageStatus = useContext(ApplicationPageStatusContext);
+
+  useLayoutEffect(() => {
+    const statusViewId = id.current;
+
+    applicationPageStatus.show(statusViewId, {
+      buttonAttrs,
+      message,
+      title,
+      variant,
+    });
+
+    return () => {
+      applicationPageStatus.hide(statusViewId);
+    };
+  }, [buttonAttrs, message, title, variant, id, applicationPageStatus]);
```

Comment:
You may consider splitting the show/hide into separate effects. Really, the only time you want to hide the overlay is if it completely unmounts. You'd just need to make sure that your provider can appropriately handle multiple `show` calls for the same id without manipulating the registration order.

Reply:
```jsx
  useLayoutEffect(() => {
    applicationPageStatus.show(id.current, {
      buttonAttrs,
      message,
      title,
      variant,
    });
  }, [buttonAttrs, message, title, variant, id, applicationPageStatus]);

  useLayoutEffect(() => () => {
    applicationPageStatus.hide(id.current);
  }, []);
```

Reply:
Addressed here : https://github.com/cerner/terra-application/pull/47/commits/108d8ce2f7edd49a0a770fdb02a18227e2239c77